remappers/HumanDynamicsRemapper/HumanDynamicsRemapper.cpp

```diff
@@ -0,0 +1,152 @@
+// parsing the configuration file and connect ports
+bool HumanDynamicsRemapper::open(yarp::os::Searchable& config)
+{
+    // ===============================
+    // CHECK THE CONFIGURATION OPTIONS
+    // ===============================
+
+    // Data ports
+    // TODO: where to check this port?
+    if (!(config.check("humanDynamicsDataPort") && config.find("humanDynamicsDataPort").isString())) {
+        yError() << logPrefix << "humanDynamicsData option does not exist or it is not a list";
+        return false;
+    }
+
+    // ===============================
+    // PARSE THE CONFIGURATION OPTIONS
+    // ===============================
+
+    std::string humanDynamicsDataPortName = config.find("humanDynamicsDataPort").asString();
+
+    // Initialize the network
+    // TODO: is this required in every DeviceDriver?
+    pImpl->network = yarp::os::Network();
+    if (!yarp::os::Network::initialized() || !yarp::os::Network::checkNetwork(5.0)) {
+        yError() << logPrefix << "YARP server wasn't found active.";
+        return false;
+    }
+
+    // ==========================
+    // CONFIGURE INPUT DATA PORTS
+    // ==========================
+    yDebug() << logPrefix << "Configuring input data ports";
+
+    pImpl->inputPort.useCallback(*this);
+    if (!pImpl->inputPort.open("...")) {
+        yError() << logPrefix << "Failed to open port" << humanDynamicsDataPortName;
+        return false;
+    }
+
+    // ================
+    // OPEN INPUT PORTS
+    // ================
+    yDebug() << logPrefix << "Opening input ports";
+
+
+    if (!yarp::os::Network::connect(humanDynamicsDataPortName,
+                                    pImpl->inputPort.getName())) {
+        yError() << logPrefix << "Failed to connect " << humanDynamicsDataPortName
+                 << " with " << pImpl->inputPort.getName();
+        return false;
+    }
+
+    // We use callbacks on the input ports, the loop is a no-op
+    start();
+
+    yDebug() << logPrefix << "Opened correctly";
+    return true;
+}
+
+void HumanDynamicsRemapper::threadRelease()
+{}
+
+bool HumanDynamicsRemapper::close()
+{
+    pImpl->terminationCall = true;
+
+    while(isRunning()) {
+        stop();
+    }
+
+    return true;
+}
+
+void HumanDynamicsRemapper::run()
+{
+    return;
+}
+
+// data are read from the port and saved in buffer variables
+void HumanDynamicsRemapper::onRead(hde::msgs::HumanDynamics& humanDynamicsData)
+{
+    if(!pImpl->terminationCall) {
+        pImpl->jointNames = humanDynamicsData.jointNames;
+
+        pImpl->jointTorques = humanDynamicsData.torques;
+    }
+}
+
+// method of IHumanDynamics interface expose the buffer variables data
+std::vector<std::string> HumanDynamicsRemapper::getJointNames() const
+{
+    return pImpl->jointNames;
+}
+
+size_t HumanDynamicsRemapper::getNumberOfJoints() const
+{
+    return pImpl->jointTorques.size();
+}
+
+std::vector<double> HumanDynamicsRemapper::getJointTorques() const
+{
+    return pImpl->jointTorques;
+}
```

Comment:
Since these methods work on shared memory, you should protect the data access with a mutex to avoid concurrency.

Reply:
> Since these methods work on shared memory, you should protect the data access with a mutex to avoid concurrency.

Ah I just saw the comment. Good suggestion! Btw should we also use the mutex in HumanStateRemapper.cpp file? Since I think the methonds defined there also share the  memory.

Reply:
> Btw should we also use the mutex in HumanStateRemapper.cpp file? Since I think the methonds defined there also share the memory.

Yes, it looks like it is missing also there. If it is quick for you, I think you could take care of the change on the `HumanStateRemapper` within this PR. Otherwise, we should open an issue on this, and do it later.